### PR TITLE
testing strat report section

### LINKED ISSUE #24: Test strategy
**Describe what needs to get done**

An explanation of how you will test the developed features of your system (in accordance with the tech stack chosen) as well as the method you will adopt to ensure continuous integration. Don't just use a technical term to describe the method; write a few sentences to explain the tools and techniques that will be involved. Read up on regression testing, and develop a feasiable plan that ensures when new code is added to your project that your project didn't break anything that was working before.

**Describe the solution you'd like**

This should be included in final report.
**Additional context**

### PR DESCRIPTION
**Pull request by edouarde1**
closes #24 and #55

### CODE REVIEW

File: docs/reports/requirements/6_test-strat.md [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
-## Test Strategy - GitHub Actions
+## Unit Testing
 
-- GitHub workflows will be created to build and test scripts
-- Frequently up-to-date YAML files that include build and test scripts will be included in GitHub repository
-- Tests will run a Node JS build
-- Triggers on GitHub workflow scripts will be triggered during each new pull request to ensure continuous integration process is followed
-- Test report will be provided on each new PR. All tests must pass in order to merge PR
+Unit tests will be built for each newly developed feature. Test scripts will be built and run using the Jest framework for Node.js and TypeScript. Within each testing script, features will be broken down and tested accordingly. As of right now the plan is to have all unit testing scripts added into a single testing directory, although script organization may change as the project develops.
```
Comment (LoganParker) [inline docs/reports/requirements/6_test-strat.md:5]: Looks good! But remember the 120 character limit for Markdown!!

**Review by LoganParker — APPROVED** [review]
These look really good! thank you :)

File: docs/reports/requirements/6_test-strat.md [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+Unit tests will be built for each newly developed feature. Test scripts will be built and run using the Jest framework for Node.js and TypeScript. Within each testing script, features will be broken down and tested accordingly. As of right now the plan is to have all unit testing scripts added into a single testing directory, although script organization may change as the project develops. 
 
+## Continuous Integration
+
+Continuous integration will be implemented using the CI/CD GitHub workflows. These workflows will be built to run tests and builds in a Node.Js environment. Initial CI testing will be focused on testing Node.Js package builds and will be ensuring all new package dependencies are accounted for in new code changes. These workflows will be triggered during each new pull request. If the tests fail, then the PR will need to be re-polished, if these tests pass then new code can be merged into our repository. We may be integrating a framework called ‘SonarCloud’ which performs automated testing and code checking, although this software may require a fee. Continuous integration workflows will be frequently updated to ensure testing scripts are up-to-date.
```
Comment (vnajack) [inline docs/reports/requirements/6_test-strat.md:9]:
```suggestion
Continuous integration will be implemented using the CI/CD GitHub workflows. These workflows will be built to run tests and builds in a Node.Js environment. Initial CI testing will be focused on testing Node.Js package builds and will be ensuring all new package dependencies are accounted for in new code changes. These workflows will be triggered during each new pull request. If the tests fail, then the PR will need to be re-polished. If these tests pass, then new code can be merged into our repository. We may be integrating a framework called ‘SonarCloud’ which performs automated testing and code checking, although this software may require a fee. Continuous integration workflows will be frequently updated to ensure testing scripts are up-to-date.
```

File: docs/reports/requirements/6_test-strat.md [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+
+## API Testing
+
+This project also requires an API build which will require separate testing to ensure that our product integrates with the client’s system. Postman will be the framework used to build and run tests for our API. This framework allows mock client requests to be made directly to our API. This framework will also ensure that our API is delivering contents correctly.
```
Comment (vnajack) [inline docs/reports/requirements/6_test-strat.md:13]:
```suggestion
This project also requires an API build which will require separate testing to ensure that our product integrates with the client’s system. Postman will be the framework used to build and run tests for our API. This framework allows mock client requests to be made directly to our API, and it ensures that our API is delivering contents correctly.
```

**Review by vnajack — CHANGES_REQUESTED** [review]
Very clear and concise! Added a couple small suggestions. Thank you

**Review by vnajack — APPROVED** [review]
Thank you so much! Looks great

**Review by mkuelker — APPROVED** [review]
Looks good

**Comment by vnajack** [comment]
Merging on behalf of Edouard - he's out of town, so was unable to merge tonight

**Pull request merged**